server/src/main/java/org/opensearch/telemetry/tracing/listener/TraceableActionListener.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.listener;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.telemetry.tracing.SpanScope;
+
+import java.util.Objects;
+
+/**
+ * Tracer wrapped {@link ActionListener}
+ * @param <Response> response.
+ */
+public class TraceableActionListener<Response> implements ActionListener<Response> {
+
+    private final ActionListener<Response> delegate;
+    private final SpanScope spanScope;
+
+    /**
+     * Constructor.
+     * @param delegate delegate
+     * @param spanScope span
+     */
+    public TraceableActionListener(ActionListener<Response> delegate, SpanScope spanScope) {
+        Objects.requireNonNull(delegate);
+        Objects.requireNonNull(spanScope);
+        this.delegate = delegate;
+        this.spanScope = spanScope;
+    }
+
+    @Override
+    public void onResponse(Response response) {
+        try (spanScope) {
```

Comment:
just use `spanScope.close()`? no need to have empty `try` blocks

Reply:
My bad. Was debugging and left this change like this only.